src/1-util.lisp

```diff
@@ -5,12 +5,13 @@
 Most arguments are analogous to mktemp.
 When DIRECTORY is non-nil, creates a directory instead.
 When DEBUG is non-nil, it does not remove the directory so that you can investigate what happened inside the directory."
-  `(let ((,var (uiop:run-program (format nil "mktemp --tmpdir='~a' ~@[-d~*~] ~a" ,tmpdir ,directory ,template)
-                                 :output '(:string :stripped t))))
+  (declare (ignorable template tmpdir))
+  `(let ((,var (uiop:run-program
+		 #-darwin (format nil "mktemp --tmpdir='~a' ~@[-d~*~] ~a" ,tmpdir ,directory ,template)
+		 #+darwin (if ,directory "mktemp -d" "mktemp")
```

Comment:
Thank you for pointing this out. It is really hard for me to notice darwin-specific changes as I don't have a Mac.
My question is: Perhaps this is not only applicable to darwin but to BSD in general? Does lisp on OSX also add BSD?
Perhaps I should also consider making it depend on `trivial-features` as it seems not all implementations add `:darwin` feature on OSX.

Reply:
Hi... unfortunately I don't have a working FreeBSD box right now but I would say what you guessed should be true, because from `man mktemp` I see clearly that this command is from BSD:

```
MKTEMP(1)                 BSD General Commands Manual                MKTEMP(1)

NAME
     mktemp -- make temporary file name (unique)

SYNOPSIS
     mktemp [-d] [-q] [-t prefix] [-u] template ...
     mktemp [-d] [-q] [-u] -t prefix
```

On the other side, yes, you should really depend on `trivial-features` for OS-specific conditional code. In this way, your code is automatically portable to unknown CL platforms.

--Chun

Reply:
For your question (Does lisp on OSX also add BSD?), the answer is NO.   SBCL does add `BSD` on Mac, while LispWorks doesn't (but once I loaded `trivial-features` I see `BSD` now).